### PR TITLE
Updated .NET Core topic with latest version numbers

### PR DESCRIPTION
**Pull request by ahardin-rh**

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@omajid @jerboaa PTAL Thanks!

File: using_images/s2i_images/dot_net_core.adoc [hunk 1]
```diff
@@ -33,21 +33,22 @@ and Python from within {product-title}.
...
 * .NET Core version 1.0
 * .NET Core version 1.1
 * Supported on Red Hat Enterprise Linux (RHEL) 7
 ifdef::openshift-enterprise[]
 and {product-title} versions 3.3 and later
 endif::openshift-enterprise[]
 
-The .NET Core software collection (`rh-dotnetcore10` and `rh-dotnetcore11`)
-ships with the *_project.json_* build system (`1.0.0-preview2` SDK). See the
+The .NET Core software collection (`rh-dotnetcore20`)
```
Comment (omajid) [inline using_images/s2i_images/dot_net_core.adoc]: `rh-dotnetcore20` ->`rh-dotnet20`

File: using_images/s2i_images/dot_net_core.adoc [hunk 2]
```diff
@@ -217,10 +215,10 @@ $ oc new-app --template dotnet-example
 ----
 
 The link:https://github.com/redhat-developer/s2i-dotnetcore-ex[.NET Core sample
-application] running on `dotnet/dotnetcore-11-rhel7` can be deployed with:
+application] running on `dotnet/dotnetcore-20-rhel7` can be deployed with:
```
Comment (omajid) [inline using_images/s2i_images/dot_net_core.adoc]: `dotnet/dotnetcore-20-rhel7` -> `dotnet/dotnet-20-rhel7`

**Review by omajid — CHANGES_REQUESTED** [review]
Should we leave instructions/steps for 1.0 and 1.1 as they are? I know that we will suggest everyone to start with 2.0 rather than 1.0 or 1.1. But is that enough of a reason to remove all steps/docs for 1.0 and 1.1 releases?

File: using_images/s2i_images/dot_net_core.adoc [hunk 3]
```diff
@@ -33,21 +33,22 @@ and Python from within {product-title}.
...
 * .NET Core version 1.1
 * Supported on Red Hat Enterprise Linux (RHEL) 7
 ifdef::openshift-enterprise[]
 and {product-title} versions 3.3 and later
 endif::openshift-enterprise[]
 
-The .NET Core software collection (`rh-dotnetcore10` and `rh-dotnetcore11`)
-ships with the *_project.json_* build system (`1.0.0-preview2` SDK). See the
+The .NET Core software collection (`rh-dotnetcore20`)
+ships with the *_project.json_* build system (`2.0.0-preview2` SDK). See the
```
Comment (omajid) [inline using_images/s2i_images/dot_net_core.adoc]: 2.0 does not support `project.json`. But 1.x will continue to support it.

**Comment by omajid** [comment]
cc @aslicerh @tmds

**Comment by leswilliams44** [comment]
Hi, @ahardin-rh:

It looks like @omajid has provided some replies to what I was going to ask.

1. Should we keep 1.0 and 1.1 in because they are using the project.json files, which is different than 2.0?

2. 2.0 is now using cproj files instead of project.json.

3. The release notes cited are for the RPMs, which is fine. There are also release notes for containers.
https://access.redhat.com/documentation/en-us/net_core/2.0/html/release_notes_for_containers/

**Comment by aslicerh** [comment]
Thanks for taking care of this @ahardin-rh!
I think everyone else has pretty much addressed my concerns, but I just wanted to point out that where we talk bout 'csproj' files, while that is technically true, other .net languages (than C#) use <something>proj files (fsproj for F#, vbproj for VB).

I don't know that we recognize that difference in any other documentation, but I thought I'd just inject that in case it is relevant.

**Comment by ahardin-rh** [comment]
@omajid @leswilliams44 and all, thanks! I made updates based on your comments. Please let me know if further edits are needed. Thanks again!

**Comment by leswilliams44** [comment]
Thanks for the updates, Ashley. @omajid and @jerboaa can verify this, but I think there is a naming difference between the 1.x versions and the 2.0 version. For instance, I think 
$ docker pull registry.access.redhat.com/dotnet/dotnetcore-10-rhel7
$ docker pull registry.access.redhat.com/dotnet/dotnetcore-11-rhel7
but the "core" was dropped for 2.0.
$ docker pull registry.access.redhat.com/dotnet/dotnet-20-rhel7

File: using_images/s2i_images/dot_net_core.adoc [hunk 4]
```diff
@@ -33,31 +33,40 @@ and Python from within {product-title}.
...
 ifdef::openshift-enterprise[]
 and {product-title} versions 3.3 and later
 endif::openshift-enterprise[]
 
-The .NET Core software collection (`rh-dotnetcore10` and `rh-dotnetcore11`)
-ships with the *_project.json_* build system (`1.0.0-preview2` SDK). See the
-Known Issues chapter in the .NET Core
-link:https://access.redhat.com/documentation/en/net-core/1.1/paged/release-notes/[Release
-Notes] for details on installing this SDK on a non-RHEL system.
+The .NET Core software collection for versions 1.1 and 1.0 (`rh-dotnet11` and
```
Comment (jerboaa) [inline using_images/s2i_images/dot_net_core.adoc]: This needs to stay at `rh-dotnetcore11`. Same for `rh-dotnetcore10`.

File: using_images/s2i_images/dot_net_core.adoc [hunk 5]
```diff
@@ -33,31 +33,40 @@ and Python from within {product-title}.
...
 
 [[dot-net-core-installing-images]]
 == Images
 
 The RHEL 7 images are available through the Red Hat Registry:
 
 ----
-$ docker pull registry.access.redhat.com/dotnet/dotnetcore-10-rhel7
-$ docker pull registry.access.redhat.com/dotnet/dotnetcore-11-rhel7
+$ docker pull registry.access.redhat.com/dotnet/dotnet-10-rhel7
```
Comment (jerboaa) [inline using_images/s2i_images/dot_net_core.adoc]: `dotnetcore-10-rhel7`

File: using_images/s2i_images/dot_net_core.adoc [hunk 6]
```diff
@@ -33,31 +33,40 @@ and Python from within {product-title}.
...
+$ docker pull registry.access.redhat.com/dotnet/dotnet-11-rhel7
```
Comment (jerboaa) [inline using_images/s2i_images/dot_net_core.adoc]: `dotnetcore-11-rhel7`

File: using_images/s2i_images/dot_net_core.adoc [hunk 7]
```diff
@@ -171,13 +180,15 @@ ifdef::openshift-online[]
...
 $ oc new-app dotnet:1.1~https://github.com/redhat-developer/s2i-dotnetcore-ex#dotnetcore-1.1 --context-dir=app
+$ oc new-app dotnet:2.0~https://github.com/redhat-developer/s2i-dotnetcore-ex#dotnetcore-2.0 --context-dir=app
 ----
 endif::openshift-online[]
 
 ifndef::openshift-online[]
 ----
-$ oc new-app registry.access.redhat.com/dotnet/dotnetcore-10-rhel7~https://github.com/redhat-developer/s2i-dotnetcore-ex#dotnetcore-1.0 --context-dir=app
-$ oc new-app registry.access.redhat.com/dotnet/dotnetcore-11-rhel7~https://github.com/redhat-developer/s2i-dotnetcore-ex#dotnetcore-1.1 --context-dir=app
+$ oc new-app registry.access.redhat.com/dotnet/dotnet-10-rhel7~https://github.com/redhat-developer/s2i-dotnetcore-ex#dotnetcore-1.0 --context-dir=app
```
Comment (jerboaa) [inline using_images/s2i_images/dot_net_core.adoc]: `oc new-app registry.access.redhat.com/dotnet/dotnetcore-10-rhel7~https://github.com/redhat-developer/s2i-dotnetcore-ex#dotnetcore-1.0 --context-dir=app`

File: using_images/s2i_images/dot_net_core.adoc [hunk 8]
```diff
@@ -171,13 +180,15 @@ ifdef::openshift-online[]
...
+$ oc new-app registry.access.redhat.com/dotnet/dotnet-11-rhel7~https://github.com/redhat-developer/s2i-dotnetcore-ex#dotnetcore-1.1 --context-dir=app
```
Comment (jerboaa) [inline using_images/s2i_images/dot_net_core.adoc]: `oc new-app registry.access.redhat.com/dotnet/dotnetcore-11-rhel7~https://github.com/redhat-developer/s2i-dotnetcore-ex#dotnetcore-1.1 --context-dir=app`

File: using_images/s2i_images/dot_net_core.adoc [hunk 9]
```diff
@@ -210,17 +221,17 @@ endif::openshift-enterprise[]
 deploy a sample application.
 
 The link:https://github.com/redhat-developer/s2i-dotnetcore-ex[.NET Core sample
-application] running on `dotnet/dotnetcore-10-rhel7` can be deployed with:
+application] running on `dotnet/dotnet-10-rhel7` can be deployed with:
```
Comment (jerboaa) [inline using_images/s2i_images/dot_net_core.adoc]: *dotnetcore*

**Review by jerboaa — CHANGES_REQUESTED** [review]
@ahardin-rh What @leswilliams44 said. See:
https://access.redhat.com/containers/#/search/dotnet for image names.

File: using_images/s2i_images/dot_net_core.adoc [hunk 10]
```diff
@@ -33,31 +33,40 @@ and Python from within {product-title}.
 [[dot-net-core-supported-versions]]
 == Supported Versions
 
+* .NET Core version 2.0
```
Comment (tmds) [inline using_images/s2i_images/dot_net_core.adoc]: Can we order this lowest to highest? (or highest to lowest, but not lowest in the middle)

File: using_images/s2i_images/dot_net_core.adoc [hunk 11]
```diff
@@ -33,31 +33,40 @@ and Python from within {product-title}.
...
 and {product-title} versions 3.3 and later
 endif::openshift-enterprise[]
 
-The .NET Core software collection (`rh-dotnetcore10` and `rh-dotnetcore11`)
-ships with the *_project.json_* build system (`1.0.0-preview2` SDK). See the
-Known Issues chapter in the .NET Core
-link:https://access.redhat.com/documentation/en/net-core/1.1/paged/release-notes/[Release
-Notes] for details on installing this SDK on a non-RHEL system.
+The .NET Core software collection for versions 1.1 and 1.0 (`rh-dotnet11` and
+`rh-dotnet10` ) ships with the *_project.json_* build system (`1.0.0-preview2`
```
Comment (tmds) [inline using_images/s2i_images/dot_net_core.adoc]: Not fun to start with a limitation here. Let's start with 2.0 first.
Suggestion on combining the two paragraphs:
```
The .NET Core software collection for version 2.0 support the msbuild/csproj build system. Versions 1.1 and 1.0 (rh-dotnetcore11 and rh-dotnetcore10 ) ships with the obsolete project.json build system (1.0.0-preview2 SDK). See the Known Issues chapter in the version 1.1 Release Notes for details on installing this SDK on a non-RHEL system.
```

File: using_images/s2i_images/dot_net_core.adoc [hunk 12]
```diff
@@ -33,31 +33,40 @@ and Python from within {product-title}.
 [[dot-net-core-supported-versions]]
 == Supported Versions
 
+* .NET Core version 2.0
 * .NET Core version 1.0
 * .NET Core version 1.1
 * Supported on Red Hat Enterprise Linux (RHEL) 7
```
Comment (tmds) [inline using_images/s2i_images/dot_net_core.adoc]: Is it intentional that '* Supported on Red Hat Enterprise Linux (RHEL) 7' is a bullet point in this list?

**Comment by ahardin-rh** [comment]
Thanks, all! This now updated.

File: using_images/s2i_images/dot_net_core.adoc [hunk 13]
```diff
@@ -33,22 +33,27 @@ and Python from within {product-title}.
...
 endif::openshift-enterprise[]
 
-The .NET Core software collection (`rh-dotnetcore10` and `rh-dotnetcore11`)
-ships with the *_project.json_* build system (`1.0.0-preview2` SDK). See the
-Known Issues chapter in the .NET Core
-link:https://access.redhat.com/documentation/en/net-core/1.1/paged/release-notes/[Release
-Notes] for details on installing this SDK on a non-RHEL system.
+The .NET Core software collection for version 2.0 supports the
+*_msbuild/csproj_* build system. Versions 1.1 and 1.0 (`rh-dotnetcore11` and
+`rh-dotnetcore10`) ships with the obsolete *_project.json_* build system
```
Comment (leswilliams44) [inline using_images/s2i_images/dot_net_core.adoc]: I don't think we should use "obsolete" for 1.1 and 1.0. The project.json files are correct for those versions.

File: using_images/s2i_images/dot_net_core.adoc [hunk 14]
```diff
@@ -58,6 +63,7 @@ The RHEL 7 images are available through the Red Hat Registry:
 ----
 $ docker pull registry.access.redhat.com/dotnet/dotnetcore-10-rhel7
 $ docker pull registry.access.redhat.com/dotnet/dotnetcore-11-rhel7
+$ docker pull registry.access.redhat.com/dotnet/dotnet-20-rhel7
 ----
```
Comment (leswilliams44) [inline using_images/s2i_images/dot_net_core.adoc]: I noticed that the versions were listed from most current to the oldest earlier (2.0, 1.1, and 1.0). Can we reorder this list in the same way for consistency?

File: using_images/s2i_images/dot_net_core.adoc [hunk 15]
```diff
@@ -217,10 +225,10 @@ $ oc new-app --template dotnet-example
...
 
 The link:https://github.com/redhat-developer/s2i-dotnetcore-ex[.NET Core sample
-application] running on `dotnet/dotnetcore-11-rhel7` can be deployed with:
+application] running on `dotnet/dotnet-20-rhel7` can be deployed with:
```
Comment (leswilliams44) [inline using_images/s2i_images/dot_net_core.adoc]: Can we reverse the order so 2.0 is first?

File: using_images/s2i_images/dot_net_core.adoc [hunk 16]
```diff
@@ -217,10 +225,10 @@ $ oc new-app --template dotnet-example
...
 
 ----
-$ oc new-app --template dotnet-example -p DOTNET_IMAGE_STREAM_TAG=dotnet:1.1 -p SOURCE_REPOSITORY_REF=dotnetcore-1.1
+$ oc new-app --template dotnet-example -p DOTNET_IMAGE_STREAM_TAG=dotnet:2.0 -p SOURCE_REPOSITORY_REF=dotnetcore-2.0
```
Comment (leswilliams44) [inline using_images/s2i_images/dot_net_core.adoc]: Can we reverse the order so 2.0 is first?

File: using_images/s2i_images/dot_net_core.adoc (hunk 13 above)
Comment (tmds) [reply using_images/s2i_images/dot_net_core.adoc]: We can leave the word out, or use deprecated instead?

**Comment by ahardin-rh** [comment]
No revision history needed

**Comment by vikram-redhat** [comment]
Added this via: https://github.com/openshift/openshift-docs/pull/5348 to the dedicated 3.6 branch so adding the label back on.

**Pull request merged**